staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -744,7 +749,7 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 				return storage.NewInternalErrorf("unable to transform key %q: %v", kv.Key, err)
 			}
 
-			if err := appendListItem(v, data, uint64(kv.ModRevision), pred, s.codec, s.versioner, newItemFunc); err != nil {
+			if err := appendListItem(v, data, uint64(kv.ModRevision), s.groupResourceString, kv.Key, pred, s.codec, s.versioner, newItemFunc); err != nil {
```

Comment:
would it be simpler to record/log inside this if block instead of plumbing `groupResource` and `key` down into `appendListItem` just for metrics/logging?

Reply:
Yes. That's my original plan. 
But I got another suggestion from @cenkalti that even though the non-nil error that  `appendListItem()` returned can only from `obj, _, err := codec.Decode(data, nil, newItemFunc())` at this moment, it would make hard for future developers who want to add more functionality into `appendListItem()` to make another non-nil error as return.
Does it make sense?
https://github.com/kubernetes/kubernetes/pull/114376#discussion_r1088124683

Reply:
Another option would be move `decode()` from `appendListItem` to here.